components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/internal/EventHookHandlerServiceComponent.java

```diff
@@ -0,0 +1,146 @@
+
+import org.wso2.identity.webhook.common.event.handler.LoginEventHookHandler;
+import org.wso2.identity.webhook.common.event.handler.builder.LoginEventPayloadBuilder;
+import com.wso2.identity.asgardeo.event.configuration.mgt.core.service.EventConfigurationMgtService;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.identity.configuration.mgt.core.ConfigurationManager;
+import org.wso2.carbon.identity.event.handler.AbstractEventHandler;
+import org.wso2.identity.event.common.publisher.EventPublisherService;
+import org.wso2.identity.webhook.common.event.handler.constant.Constants;
+import org.wso2.identity.webhook.common.event.handler.util.EventHookHandlerUtils;
+
+/**
+ * WSO2 Event Handler service component class.
+ */
+@Component(
+        name = "org.wso2.identity.webhook.common.event.handler.internal.EventHookHandlerServiceComponent",
+        immediate = true)
+public class EventHookHandlerServiceComponent {
+
+    private static final Log log = LogFactory.getLog(EventHookHandlerServiceComponent.class);
+
+    @Activate
+    protected void activate(ComponentContext context) {
+
+        try {
+            log.debug("Event Handler is activated.");
+            String isLoginEventHandlerEnabled = EventHookHandlerUtils
+                    .getIdentityEventProperty(Constants.LOGIN_EVENT_HOOK_NAME, Constants.LOGIN_EVENT_HOOK_ENABLED);
+            BundleContext bundleContext = context.getBundleContext();
+            if (isLoginEventHandlerEnabled.equalsIgnoreCase(Boolean.TRUE.toString())) {
+                bundleContext.registerService(AbstractEventHandler.class.getName(),
+                        new LoginEventHookHandler(), null);
+            } else {
+                log.error("Login Event Handler is not enabled.");
+            }
+        } catch (Exception e) {
+            log.error("Error while activating event handler.", e);
+        }
+    }
+
+    @Deactivate
+    protected void deactivate(ComponentContext context) {
+
+        log.debug("Event Handler is deactivated.");
+    }
+
+    @Reference(
+            name = "login.event.payload.builder",
+            service = LoginEventPayloadBuilder.class,
+            cardinality = ReferenceCardinality.MULTIPLE,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "removeLoginEventPayloadBuilder"
+    )
+    protected void addLoginEventPayloadBuilder(LoginEventPayloadBuilder loginEventPayloadBuilder) {
```

Comment:
Does this needs to be protected ? Have the same comment on all methods here

Reply:
IMO, this is not a class intended to be extended

Reply:
If it's not protected, the osgi framework cannot invoke these method when activating and deactivating.